src/api/java/com/minecolonies/api/compatibility/CompatibilityManager.java

```diff
@@ -43,21 +42,26 @@
 
     /**
      * List of saplings.
+     * Works on client and server-side.
      */
     private final List<ItemStorage> saplings = new ArrayList<>();
 
     /**
      * List of all ore-like blocks.
+     * Works on client and server-side.
      */
     private final List<Block> ores = new ArrayList<>();
 
     /**
-     * The oredict entry of an ore.
+     * Instantiates the compatibilityManager.
      */
-    public static final String ORE_STRING = "ore";
+    public CompatibilityManager()
+    {
+        discover();
+    }
 
     @Override
-    public void discover(final World world)
+    public void discover()
```

Comment:
Please offload this to another thread and start at a appropriate moment. This causes else a big hangup on the main thread when it is not needed.

Reply:
We had always run this on readFromNBT, now it would be called on serverstart as well + on client start, would that be much of an issue?